spring/boot2-autoconfigure/src/test/resources/config/application-settings.yml

```diff
@@ -0,0 +1,17 @@
+armeria:
+  max-num-connections: 8
+  idle-timeout: 2s
+  ping-interval: 1s
+  max-connection-age: 4s
+  max-num-requests-per-connection: 4
+  http2-initial-connection-window-size: 2mb
+  http2-initial-stream-window-size: 4mb
+  http2-max-streams-per-connection: 1
+  http2-max-frame-size: 32768
+  http2-max-header-list-size: 16384
+  http1-max-initial-line-length: 8192
+  http1-max-header-size: 16384
+  http1-max-chunk-size: 32768
+  request-timeout: 8000
```

Comment:
Supports `long` milliseconds by `NumberToDurationConverter`.

Reply:
I look fine 🤔

Reply:
how about providing two setters for all durations? one that ends with `-millis` that accepts a number and the one without a suffix which accepts a string with unit, e.g. `8s` or `8000ms`?

Reply:
If there are two setters, users can be confused about which one has priority. Do you worry that `request-timeout: 8000` doesn't represent unit?

Reply:
Yeah, that's my concern. When a user specifies two properties that are translated into one property, in which order does Spring Boot call the setters? If the firstly specified property is always called before the secondly specified one, I guess it's probably OK. What do you think, @ikhoon @minwoox ?

Reply:
Alternatively, we could just enforce a suffix (i.e. disallow a value without unit suffix)

Reply:
Personally, I prefer to allow no unit suffix because [spring-boot allows](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.conversion.durations):

> - A regular long representation (using milliseconds as the default unit unless a @DurationUnit has been specified)
> - The standard ISO-8601 format used by java.time.Duration
> - A more readable format where the value and the unit are coupled (e.g. 10s means 10 seconds)

And `parseDataSize` allows no unit suffix (it means byte):

https://github.com/line/armeria/blob/bcbe65800e581a1e422368b258668cc01355e27d/spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtil.java#L373-L376

However, I fully agree @trustin and @ikhoon's comments and supporting is easier than un-supporting in this case. It would be better to allow no unit suffix when really needs.
So +1 too.

Reply:
> A regular long representation (using milliseconds as the default unit unless a @DurationUnit has been specified)

TIL. If Spring Boot already supports no unit suffix, I'm good to support long representations.

Reply:
I can revert 54a8ba2. Could you confirm? I'm fine both.

Reply:
> I can revert [54a8ba2](https://github.com/line/armeria/commit/54a8ba2bdc0866e61698a62b59838a3639fdf385). Could you confirm? 

I prefer to revert it. But I'd like to listen to other folks' opinions.

Reply:
OK to revert :sweat_smile: